src/pages/index.astro

```diff
@@ -2,16 +2,22 @@
 import { Body } from "../components/index/body.tsx";
 import fetch from 'node-fetch';
 import * as cheerio from 'cheerio';
+import NetLogoApi from "../utils/api.js";
+
+const api = new NetLogoApi();
+
+const allData = await api.getAllData();
+console.log(allData);
 
 let base = 'https://backend.netlogo.org';
 
 let response = await fetch(base + '/items/announcements')
 let data: any = await response.json();
-const announcement = data.data;0
+const announcement = data.data;
```

Comment:
For all of these three liners like:
```
let response = await fetch(base + '/items/announcements')
let data: any = await response.json();
const announcement = data.data;
```
can't we create a little function that takes in a url and returns the data and then call that function for each of announcement, upcomingEvents, etc?